packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/client_config.ts

```diff
@@ -9,11 +9,12 @@
 import { ConnectionOptions as TlsConnectionOptions } from 'tls';
 import { URL } from 'url';
 import { Duration } from 'moment';
-import type { ClientOptions, HttpAgentOptions } from '@elastic/elasticsearch';
+import type { ClientOptions } from '@elastic/elasticsearch';
 import type { ElasticsearchClientConfig } from '@kbn/core-elasticsearch-server';
+import { AgentOptions } from 'https';
```

Comment:
Although Elasticsearch-js exposes an `HttpAgentOptions` type, this type is not up to date with the Nodejs type.